moped-editor/src/components/DataGridPro/DataGridActions.js

```diff
@@ -0,0 +1,96 @@
+import {
+  GridRowModes,
+  GridActionsCellItem,
+  useGridApiContext,
+  useGridSelector,
+} from "@mui/x-data-grid-pro";
+import CheckIcon from "@mui/icons-material/Check";
+import CloseIcon from "@mui/icons-material/Close";
+import {
+  EditOutlined as EditOutlinedIcon,
+  DeleteOutline as DeleteOutlineIcon,
+} from "@mui/icons-material";
+
+import { defaultEditColumnIconStyle } from "src/utils/dataGridHelpers";
+
+/** Component for Data Grid table action buttons
+ * @param {Integer} id - Data Grid row id (same as project id)
+ * @param {Array} requiredFields - fields that are required in order to save row
+ * @param {Object} rowModesModel - row modes state from data grid
+ * @param {Function} handleCancelClick - handles cancel button click
+ * @param {Function} handleDeleteClick - handles delete button click
+ * @param {Function} handleSaveClick - handles save button click
+ * @param {Function} handleEditClick - handles edit button click, optional
+ * @return {JSX.Element}
+ */
+
+const DataGridActions = ({
+  id,
+  requiredFields,
+  rowModesModel,
+  handleCancelClick,
+  handleDeleteOpen,
+  handleSaveClick,
+  handleEditClick,
+}) => {
+  const isInEditMode = rowModesModel[id]?.mode === GridRowModes.Edit;
+
+  const apiRef = useGridApiContext();
+
+  /**
+   * To make our row re-render while still in edit mode we need the useGridSelector hook
+   * which establishes a reactive binding with the grid state and allows us to enable the save button
+   * if we have all the required fields.
+   * For reference https://mui.com/x/react-data-grid/state/#access-the-state
+   */
+  const hasRequiredFields = useGridSelector(apiRef, () => {
+    const editState = apiRef.current.state.editRows;
+    for (const field of requiredFields) {
+      if (!editState[id]?.[field]?.value) {
+        return false;
+      }
+    }
+    return true;
+  });
+
+  if (isInEditMode) {
+    return [
+      <GridActionsCellItem
+        icon={<CheckIcon sx={defaultEditColumnIconStyle} />}
+        label="Save"
+        sx={{
+          color: "primary.main",
+        }}
+        onClick={handleSaveClick(id)}
+        disabled={!hasRequiredFields}
+      />,
+      <GridActionsCellItem
+        icon={<CloseIcon sx={defaultEditColumnIconStyle} />}
+        label="Cancel"
+        className="textPrimary"
+        onClick={handleCancelClick(id)}
+        color="inherit"
+      />,
+    ];
+  }
+  return [
+    // only render edit button if we were passed an edit handler and are not currently in edit mode
+    handleEditClick && (
+      <GridActionsCellItem
+        icon={<EditOutlinedIcon sx={defaultEditColumnIconStyle} />}
+        label="Edit"
+        className="textPrimary"
+        onClick={handleEditClick(id)}
+        color="inherit"
+      />
+    ),
+    <GridActionsCellItem
+      icon={<DeleteOutlineIcon sx={defaultEditColumnIconStyle} />}
+      label="Delete"
+      onClick={() => handleDeleteOpen(id)}
```

Comment:
This is not related to code introduced in this PR, but seeing this anonymous function for this delete handler right next to function calls of `handleCancelClick(id)` and `handleEditClick(id)` makes me feel like we should update the definition of the delete handlers in the table components to match up with the [MUI code example](https://mui.com/x/react-data-grid/editing/#full-featured-crud) syntax. [This thread](https://austininnovation.slack.com/archives/CHZE6BC6L/p1730494043545059) is also related for context. like:

```js
const handleDeleteOpen = (id) => () => {
    // do something with id
};
```

Would you up for making this change in this PR if it feels small to you? I'm happy to make a snackoo too since its performance impact isn't immediate.

Reply:
thanks for squeezing this in! 🙌